sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java

```diff
@@ -0,0 +1,818 @@
+       * <code>optional int32 start_offset = 3;</code>
+       *
+       * <pre>
+       * If absent it means range starts at the first offset of start line
+       * </pre>
+       */
+      public Builder clearStartOffset() {
+        bitField0_ = (bitField0_ & ~0x00000004);
+        startOffset_ = 0;
+        onChanged();
+        return this;
+      }
+
+      private int endOffset_ ;
+      /**
+       * <code>optional int32 end_offset = 4;</code>
+       *
+       * <pre>
+       * If absent it means range ends at the last offset of end line
+       * </pre>
+       */
+      public boolean hasEndOffset() {
+        return ((bitField0_ & 0x00000008) == 0x00000008);
+      }
+      /**
+       * <code>optional int32 end_offset = 4;</code>
+       *
+       * <pre>
+       * If absent it means range ends at the last offset of end line
+       * </pre>
+       */
+      public int getEndOffset() {
+        return endOffset_;
+      }
+      /**
+       * <code>optional int32 end_offset = 4;</code>
+       *
+       * <pre>
+       * If absent it means range ends at the last offset of end line
+       * </pre>
+       */
+      public Builder setEndOffset(int value) {
+        bitField0_ |= 0x00000008;
+        endOffset_ = value;
+        onChanged();
+        return this;
+      }
+      /**
+       * <code>optional int32 end_offset = 4;</code>
+       *
+       * <pre>
+       * If absent it means range ends at the last offset of end line
+       * </pre>
+       */
+      public Builder clearEndOffset() {
+        bitField0_ = (bitField0_ & ~0x00000008);
+        endOffset_ = 0;
+        onChanged();
+        return this;
+      }
+
+      // @@protoc_insertion_point(builder_scope:sonarqube.db.commons.TextRange)
+    }
+
+    static {
+      defaultInstance = new TextRange(true);
+      defaultInstance.initFields();
+    }
+
+    // @@protoc_insertion_point(class_scope:sonarqube.db.commons.TextRange)
+  }
+
+  private static final com.google.protobuf.Descriptors.Descriptor
+    internal_static_sonarqube_db_commons_TextRange_descriptor;
+  private static
+    com.google.protobuf.GeneratedMessage.FieldAccessorTable
+      internal_static_sonarqube_db_commons_TextRange_fieldAccessorTable;
+
+  public static com.google.protobuf.Descriptors.FileDescriptor
+      getDescriptor() {
+    return descriptor;
+  }
+  private static com.google.protobuf.Descriptors.FileDescriptor
+      descriptor;
+  static {
+    java.lang.String[] descriptorData = {
+      "\n\020db-commons.proto\022\024sonarqube.db.commons" +
+      "\"[\n\tTextRange\022\022\n\nstart_line\030\001 \001(\005\022\020\n\010end" +
+      "_line\030\002 \001(\005\022\024\n\014start_offset\030\003 \001(\005\022\022\n\nend" +
+      "_offset\030\004 \001(\005B\031\n\025org.sonar.db.protobufH\001"
+    };
+    com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner assigner =
+        new com.google.protobuf.Descriptors.FileDescriptor.    InternalDescriptorAssigner() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)